src/index.ts

```diff
@@ -1006,12 +1039,36 @@ class Envoy {
     return resp;
   }
 
+  async getServiceLoggerCellId(hha_hash) {
+    let servicelogger_installed_app_id;
+
+    if (this.opts.hosted_app && this.opts.hosted_app!.servicelogger_id && this.opts.mode === Envoy.DEVELOP_MODE) {
+      servicelogger_installed_app_id = this.opts.hosted_app.servicelogger_id;
+    } else {
+      // NB: There will be a new servicelogger app for each hosted happ (should happen at the time of self-hosted install - prompted in host console.)
+      servicelogger_installed_app_id = `${hha_hash}::servicelogger`;
+    }
+
+    log.info("Retrieve Servicelogger cell id using the Installed App Id: '%s'", servicelogger_installed_app_id);
+    const appInfo = await this.callConductor("app", { installed_app_id: servicelogger_installed_app_id });
+
+    if (!appInfo) {
+      log.error("Failed during Servicelogger AppInfo lookup: %s", appInfo);
+      throw new HoloError("Failed to fetch AppInfo for Servicelogger")
+    }
+
+    log.debug("Servicelogger app_info: '%s'", appInfo);
+    // We are assuming that servicelogger is a happ and the first cell is the servicelogger DAN
```

Comment:
```suggestion
    // We are assuming that servicelogger is a happ and the first cell is the servicelogger DNA
```
I realize I had left a typo in this. :laughing: